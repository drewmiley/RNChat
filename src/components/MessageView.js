import React, { Component } from 'react';
import {
    Text,
    View
} from 'react-native';

export default class MessageView extends Component {
    constructor(props) {
        super(props);
        this.shouldComponentUpdate = false;
    };
    render() {
        if (this.props.drewSaid) {        
            return <Text>DREW : {this.props.text}</Text>
        } else {
            return <Text>YOU : {this.props.text}</Text>
        };
    };
};