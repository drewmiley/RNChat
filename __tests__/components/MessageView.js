import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import MessageView from '../..//src/components/MessageView';

describe('MessageView', () => {

    describe('Rendering', () => {

        it('should match the snapshot', () => {
            const component = renderer.create(
                <MessageView />
            );            
            expect(component).toMatchSnapshot();
        });

    });

});