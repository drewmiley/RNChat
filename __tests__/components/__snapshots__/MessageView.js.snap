exports[`MessageView Rendering should match the snapshot 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail">
  YOU : 
</Text>
`;
